Include/internal/pycore_interp.h

```diff
@@ -33,6 +33,7 @@ extern "C" {
 #include "pycore_typeobject.h"    // struct type_cache
 #include "pycore_unicodeobject.h" // struct _Py_unicode_state
 #include "pycore_warnings.h"      // struct _warnings_runtime_state
+#include "cpython/optimizer.h"
```

Comment:
"Reverse" includes like this set off my "bad architecture" warnings. Ideally the internal headers would only include from other internal headers, and the `cpython` ones can reach down.

If it's unavoidable, at least add a comment suggesting why (I assume it's so you can use the typedef'd `_PyOptimizerObject` below instead of `struct _PyOptimizerObject`).

Totally fine to require anyone including these headers to also include the higher level ones if they want to use the object. Shouldn't be required just to compile though.